config/archive_test.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"path"
+	"time"
+
+	"github.com/contiv/volplugin/watch"
+	"github.com/coreos/etcd/client"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testPolicyName = "foobar"
+)
+
+func revisionCount(s *configSuite, c *C) int {
+	revisions, _ := s.tlc.ListPolicyRevisions(testPolicyName)
+	return len(revisions)
+}
+
+func createTestPolicy(s *configSuite, c *C) {
+	err := s.tlc.CreatePolicyRevision(testPolicyName, "{}")
+	c.Assert(err, IsNil)
+}
+
+func (s *configSuite) TestCreatePolicyRevision(c *C) {
+	c.Assert(revisionCount(s, c), Equals, 0)
+	createTestPolicy(s, c)
+	c.Assert(revisionCount(s, c), Equals, 1)
+}
+
+// the logic for testing Get and List only differs in that testing Get verifies
+// that the policy text is what was uploaded, so the two tests are combined here.
+func (s *configSuite) TestGetAndListPolicyRevisions(c *C) {
+	policyText := `{"foo":"bar"}`
+
+	c.Assert(revisionCount(s, c), Equals, 0)
+
+	err := s.tlc.CreatePolicyRevision(testPolicyName, policyText)
+	c.Assert(err, IsNil)
+
+	revisions, err := s.tlc.ListPolicyRevisions(testPolicyName)
+	c.Assert(err, IsNil)
+	c.Assert(len(revisions), Equals, 1)
+
+	resp, err := s.tlc.GetPolicyRevision(testPolicyName, revisions[0])
+	c.Assert(err, IsNil)
+	c.Assert(resp, Equals, policyText)
+}
+
+// this test ensures that a new policy being uploaded results in the correct
+// data (policy name + timestamp, tab-separated) being written to stdout
+func (s *configSuite) TestWatchForPolicyChanges(c *C) {
+	changes := [][2]string{}
```

Comment:
Done.  I was thinking in terms of a "changeset" being returned as a tuple, but, as long as we're asserting the length is correct, it doesn't really matter.